common/src/main/java/com/wynntils/wynn/model/ShamanTotemModel.java

```diff
@@ -104,6 +109,64 @@ public void onTotemSpawn(AddEntityEvent e) {
+        if (getBoundTotem(entityId) != null) return;
+
+        Entity possibleTimer = getBufferedEntity(entityId);
+        if (!(possibleTimer instanceof ArmorStand)) return;
+
+        // Given timerId is not a totem, make a new totem
+        List<ArmorStand> toCheck = McUtils.mc()
+                .level
+                .getEntitiesOfClass(
+                        ArmorStand.class,
+                        new AABB(
+                                possibleTimer.position().x - TOTEM_SEARCH_RADIUS,
+                                possibleTimer.position().y
+                                        - 0.3, // Don't modify this unless you are certain it is causing issues
+                                possibleTimer.position().z - TOTEM_SEARCH_RADIUS,
+                                possibleTimer.position().x + TOTEM_SEARCH_RADIUS,
+                                // (a LOT) more vertical radius required for totems casted off high places
+                                // This increased radius requirement increases as you cast from higher places and as the
+                                // server gets laggier
+                                possibleTimer.position().y + TOTEM_SEARCH_RADIUS * 5,
+                                possibleTimer.position().z + TOTEM_SEARCH_RADIUS));
+
+        for (ArmorStand as : toCheck) {
+            // Recreate location for each ArmorStand checked for most accurate coordinates
+            Location parsedLocation = new Location(as.position().x, as.position().y, as.position().z);
+            if (pendingTotem1VisibleId != null && as.getId() == pendingTotem1VisibleId) {
+                totem1.setTimerEntityId(entityId);
+                totem1.setLocation(parsedLocation);
+                totem1.setState(ShamanTotem.TotemState.ACTIVE);
+                WynntilsMod.postEvent(new TotemEvent.Activated(1, parsedLocation));
+                pendingTotem1VisibleId = null;
+            } else if (pendingTotem2VisibleId != null && as.getId() == pendingTotem2VisibleId) {
+                totem2.setTimerEntityId(entityId);
+                totem2.setLocation(parsedLocation);
+                totem2.setState(ShamanTotem.TotemState.ACTIVE);
+                WynntilsMod.postEvent(new TotemEvent.Activated(2, parsedLocation));
+                pendingTotem2VisibleId = null;
+            } else if (pendingTotem3VisibleId != null && as.getId() == pendingTotem3VisibleId) {
+                totem3.setTimerEntityId(entityId);
+                totem3.setLocation(parsedLocation);
+                totem3.setState(ShamanTotem.TotemState.ACTIVE);
+                WynntilsMod.postEvent(new TotemEvent.Activated(3, parsedLocation));
+                pendingTotem3VisibleId = null;
+            } else {
+                // No totem slots available?
+                // WynntilsMod.getLogger().warn("Received a new totem {}, but no totem slots are available", entityId);
```

Comment:
```suggestion
                WynntilsMod.getLogger().warn("Received a new totem {}, but no totem slots are available", entityId);
```

Reply:
I want to keep this as a comment, as the warn is useful for debugging purposes, but at the moment, the server sends a duplicate packet for each totem. So even if we successfully capture the totem, we will get an incorrect warn for every totem casted. If the duplicate packet disappears in the future, we can uncomment again.

Reply:
ok